src/settings.js

```diff
@@ -2,44 +2,76 @@ import { isPlainObject } from './helpers';
 import { Value } from './values';
 
 export default class Settings {
-  static prefix;
-
-  constructor(settings) {
-    this.cache = new Map();
-    this.proxy = new Proxy(settings, {
-      get: (target, key) => {
-        if (this.cache.has(key)) {
-          return this.cache.get(key);
-        }
+  static assign(...args) {
+    return new Settings(
+      Object.assign(
+        ...args.map(arg => {
+          if (arg instanceof Settings) {
+            const settings = arg._settings;
+            Object.entries(settings).forEach(([key, value]) => {
+              if (value instanceof Value && value.options.envPrefix === undefined) {
+                settings[key].options.envPrefix = arg._prefix;
+              }
+            });
+            return settings;
+          }
+          return arg;
+        })
+      )
+    );
+  }
 
-        const value = this.parseValue(target[key], key);
-        this.cache.set(key, value);
-        return value;
+  constructor(settings = {}, prefix) {
+    const cache = new Map();
+
+    // We use defineProperties because we don't want these to be enumerable
+    Object.defineProperties(this, {
+      _settings: {
+        value: { ...settings },
+      },
+      _prefix: {
+        value: prefix,
       },
     });
 
-    return this.proxy;
-  }
-
-  parseValue(value, envName) {
-    if (typeof value === 'function') {
-      value = value.apply(this.proxy);
-    }
-
-    if (value instanceof Value) {
-      if (value.options.envName === undefined) {
-        value.options.envName = envName;
+    const evaluate = (value, envName) => {
```

Comment:
`envName` isn't needed anymore, is it?

Reply:
it is. it's passed through on: https://github.com/rehandalal/node-env-settings/pull/4/files/82cd050118c5a4b86c356d51a386d23282dd4bb1#diff-43d6bd0c5e0f9bc940befa68ab214b4dR67

that line however needs to be updated to remove the `prefix`.